### PR TITLE
Support for adding secrets

### PR DESCRIPTION
**Pull request by MicroFish91**
Demo:

### CODE REVIEW

File: src/commands/secret/addSecret/SecretValueStep.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        });
+        context.valuesToMask.push(context.secretValue);
+    }
+
+    public shouldPrompt(context: ISecretContext): boolean {
+        return !context.secretValue;
+    }
+
+    private validateInput(val: string | undefined): string | undefined {
+        val = val ? val.trim() : '';
```
Comment (bwateratmsft) [inline src/commands/secret/addSecret/SecretValueStep.ts:26]: Should the secret be trimmed, even for validation? A lot of services don't work with a secret that begins or ends with whitespace but IMO that's a bug.

File: src/utils/validateUtils.ts [hunk 2]
```diff
@@ -40,4 +41,30 @@ export namespace validateUtils {
...
+    /**
+     * Validates that the given input string consists of lower case alphanumeric characters,
+     * starts and ends with an alphanumeric character, and does not contain any special symbols not explicitly specified
+     *
+     * @param value The original input string to validate
+     * @param symbols Any custom symbols that are also allowed in the input string. Defaults to '-'.
+     *
+     * @example
+     * "abcd-1234" // returns true
+     * "-abcd-1234" // returns false
```
Comment (bwateratmsft) [inline src/utils/validateUtils.ts:54]: Maybe some unit tests are in order for this?

File: src/utils/validateUtils.ts [hunk 3]
```diff
@@ -40,4 +41,30 @@ export namespace validateUtils {
...
+     */
+    export function isLowerCaseAlphanumericWithSymbols(value: string, symbols: string = '-'): boolean {
```
Comment (MicroFish91) [inline src/utils/validateUtils.ts:56]: Was thinking we could expand on this with additional options in the future as well.  For example: (1) option to allow uppercase values (would have to rename the function), (2) option to restrict repeating symbols, etc.

Could also add these to shared utils if others find this useful in some capacity.  Open to any thoughts/recommendations :)

File: src/commands/secret/ISecretContext.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import type { Secret } from "@azure/arm-appcontainers";
+import type { ExecuteActivityContext } from "@microsoft/vscode-azext-utils";
+import type { IContainerAppContext } from "../IContainerAppContext";
+
+export interface ISecretContext extends IContainerAppContext, ExecuteActivityContext {
+    // New
+    secretName?: string;
```
Comment (nturinski) [inline src/commands/secret/ISecretContext.ts:12]: I think it would be easier to just name the variable names `newSecretName` and `newSecretValue`.

**Review by nturinski — APPROVED** [review]

**Pull request merged**